crates/fuzzing/src/oracles/diff_wasmtime.rs

```diff
@@ -0,0 +1,184 @@
+//! Evaluate an exported Wasm function using Wasmtime.
+
+use crate::generators::{self, DiffValue, InstanceAllocationStrategy, ModuleFeatures};
+use crate::oracles::engine::DiffInstance;
+use crate::oracles::{compile_module, engine::DiffEngine, instantiate_with_dummy, StoreLimits};
+use anyhow::{Context, Result};
+use arbitrary::Unstructured;
+use std::cell::RefCell;
+use std::hash::Hash;
+use std::ops::{Deref, DerefMut};
+use std::slice;
+use wasmtime::{Extern, Instance, Store, Val};
+
+/// A wrapper for using Wasmtime as a [`DiffEngine`].
+pub struct WasmtimeEngine {
+    config: generators::Config,
+}
+
+impl WasmtimeEngine {
+    /// Merely store the configuration; the engine is actually constructed
+    /// later. Ideally the store and engine could be built here but
+    /// `compile_module` takes a [`generators::Config`]; TODO re-factor this if
+    /// that ever changes.
+    pub fn new(config: &generators::Config) -> Result<Box<Self>> {
+        Ok(Box::new(Self {
+            config: config.clone(),
+        }))
+    }
+
+    /// Create a new Wasmtime engine from a randomly-generated configuration
+    /// that matches the given Wasm `features`.
+    pub fn arbitrary_with_features(
```

Comment:
Personally I get pretty easily lost in the maze of `Arbitrary` implementations and methods of configuration. Would it be possible to avoid having this and `arbitrary_with_compatible_config` below? Could some higher-up generate the `Config`/`Engine` and be responsible for passing it down?

Or otherwise it seems like with Wasmtime we'd want a constant `ModuleConfig` with possibly differing `WasmtimeConfig` or something like that.

If this is too difficult though or otherwise requires complications then I think that removing differential-wasmtime-with-itself is pretty reasonable. 90% of the value comes from differential execution with _something_ so adding more engines IMO doesn't increase our coverage really all that much.

Reply:
Although as I say that I remember that different engines implement different features which is often the hard part and nothing implements Wasmtime's feature set other than Wasmtime exactly.